### ISSUE TITLE
[SignalR] Java Client NullPointerException

### ISSUE THREAD

**Issue by JMaroz** [aspnet/AspNetCore#6898]
### Describe the bug
When start the connection the SignalR HubConnection class throw an Exception

`Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.microsoft.signalr.HubConnection$ConnectionState.cancelOutstandingInvocations(java.lang.Exception)' on a null object reference
       at com.microsoft.signalr.HubConnection.stopConnection(HubConnection.java:6)
       at com.microsoft.signalr.HubConnection.lambda$start$6(HubConnection.java:1)
       at com.microsoft.signalr.s.invoke(Unknown Source:2)
       at com.microsoft.signalr.WebSocketTransport.onClose(WebSocketTransport.java:4)
       at com.microsoft.signalr.WebSocketTransport.lambda$start$1(WebSocketTransport.java:3)
       at com.microsoft.signalr.A.invoke(Unknown Source:2)
       at com.microsoft.signalr.OkHttpWebSocketWrapper$SignalRWebSocketListener.onFailure(OkHttpWebSocketWrapper.java:3)
       at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:84)
       at okhttp3.internal.ws.RealWebSocket$2.onFailure(RealWebSocket.java:1)
       at okhttp3.RealCall$AsyncCall.execute(RealCall.java:8)
       at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:3)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)`

### To Reproduce
Unable to reproduce the behavior, the crash happen randomly

### Expected behavior
SignalR must catch the exception or not throw the exception

**Comment by jkotalik** [comment]
@mikaelm12 @BrennanConroy

**Comment by mikaelm12** [comment]
I'll investigate. Thanks!

**Comment by bradygaster** [comment]
@JMaroz are you connecting to a traditional server-side Hub or are you connecting through the SignalR Service? Just curious.

**Comment by mikaelm12** [comment]
So I believe that I was running into this bug during my work on the LongPolling transport and I fixed it in that pr which is still under review https://github.com/aspnet/AspNetCore/pull/6856. The issue I believe was 
the connection state wasn't being set until we received the Handshake response so if the connection is stopped before it has been fully set up you could null ref in stop. This should be fixed in the LongPolling pr. https://github.com/aspnet/AspNetCore/blob/a6f4a27abb2db0dcd65d1595c43bcde048b85228/src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java#L326

**Comment by imperugo** [comment]
@bradygaster let me share some information about server side. The problem explained by @JMaroz happens with this configuration:

- ASP.NET Core 2.2.0
- SignalR 1.1.0
- Hosting on Docker over ECS on AWS Cloud Platform (no reverse proxy, just kestrel behind the balancer)

The client connects to simple Hub service that looks like this (more or less)

```csharp
public class MySuperHub : HubBase
{
   public async Task DoSomethingSpecial(string param1, Guid param2, dynamic param3 = null)
   {
      // await ....
   }
}
```

Let me know if you need more info.
Thanks

**Comment by bradygaster** [comment]
@mikaelm12 and @muratg would this be ready (or is it already in) for 2.2.3?

**Comment by muratg** [comment]
@bradygaster To get this in 2.2.3, we'll need a PR with the fix with the label `servicing-consider`. It'll show up in the next shiproom then.

**Comment by mikaelm12** [comment]
I don't think that that would be critical to get into 2.2.3. It's an error that happens in a failure in the close logic of the Java client. Happy to discuss it more though if people have other opinions.

**Comment by JMaroz** [comment]
@mikaelm12 is there a release date w/ the fix for the java client?

**Comment by mikaelm12** [comment]
The fix for the null ref will be coming in Preview3, but that is not why your HubConnection fails to start. The exception you posted is the result of you HubConnection failing to start, not the cause of it I believe. Do you have any more logs or a repro app we can look at?

**Comment by mikaelm12** [comment]
The failure in the close logic was fixed here 3d3ad96206fc504dba954c0c6403139e3438263e
